docs/iris/src/sphinxext/generate_logo.py

```diff
@@ -0,0 +1,441 @@
+# Set plot to include space for the mask to be added.
+# figure_inches doesn't influence the final size, but does influence the coastline definition.
+figure_inches = 10
+mpl_points_per_inch = 72
+
+globe_inches = figure_inches / CLIP_GLOBE_RATIO
+pad_inches = (figure_inches - globe_inches) / 2
+background_points = figure_inches * mpl_points_per_inch
+globe_points = globe_inches * mpl_points_per_inch
+
+mpl_font_points = rcParams["font.size"]
+inch_font_ratio = mpl_points_per_inch / mpl_font_points
+
+# XML ElementTree setup.
+namespaces = {"svg": "http://www.w3.org/2000/svg"}
+ET.register_namespace("", namespaces["svg"])
+
+################################################################################
+# Create new SVG elements for logo.
+
+# The elements that are displayed (not just referenced).
+# Order is important for layering.
+artwork_dict = OrderedDict.fromkeys(["background", "glow", "sea", "land",])
+# The elements that will just be referenced by artwork elements.
+defs_dict = {}
+
+# BACKGROUND
+artwork_dict["background"] = ET.Element(
+    "rect",
+    attrib={
+        "height": "100%",
+        "width": "100%",
+        "fill": "url(#background_gradient)",
+    },
+)
+background_gradient = ET.Element(
+    "linearGradient", attrib={"y1": "0%", "y2": "100%",},
+)
+background_gradient.append(
+    ET.Element("stop", attrib={"offset": "0", "stop-color": "#13385d",},)
+)
+background_gradient.append(
+    ET.Element("stop", attrib={"offset": "1", "stop-color": "#272b2c",},)
+)
+defs_dict["background_gradient"] = background_gradient
+
+# LAND
+# (Using Matplotlib and Cartopy).
+
+# Create land with simplified coastlines.
+simple_geometries = [
+    geometry.simplify(1.0, True) for geometry in LAND.geometries()
+]
+LAND.geometries = lambda: iter(simple_geometries)
+
+# Variable that will store the sequence of land-shaped SVG clips for each longitude.
+land_clips = OrderedDict()
+
+# Create a sequence of longitude values.
+central_longitude = -30
+central_latitude = 22.9
+rotation_frames = 180
+rotation_longitudes = linspace(start=central_longitude + 360,
+                               stop=central_longitude,
+                               num=rotation_frames,
+                               endpoint=False)
+# Normalise to -180..+180
+rotation_longitudes = (rotation_longitudes + 360.0 + 180.0) % 360.0 - 180.0
+
+for lon in rotation_longitudes:
+    # Use Matplotlib and Cartopy to generate land-shaped SVG clips for each longitude.
+
+    projection_rotated = ccrs.Orthographic(central_longitude=lon,
+                                           central_latitude=central_latitude)
+
+    fig = plt.figure(0, figsize=(figure_inches,) * 2)
+    ax = plt.axes(projection=projection_rotated)
+    # TODO: remove version check Iris is pinned appropriately.
+    if cartopy_version < "0.18":
+        fig.canvas.draw()
+    # Use constants set earlier to achieve desired dimensions.
+    plt.tight_layout(pad=pad_inches * inch_font_ratio)
```

Comment:
It would be safer if the figure size were correct from the beginning, if possible.

Reply:
Cool, I've switched to using `plt.subplots_adjust` and that's removed a few hacky lines. Hopefully the new code is a more correct use.